common/meta/types/src/meta_raft_errors.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use openraft::error::ChangeMembershipError;
+use openraft::NodeId;
+use serde::Deserialize;
+use serde::Serialize;
+use thiserror::Error;
+
+pub type ForwardToLeader = openraft::error::ForwardToLeader;
+
+// represent raft related errors
+#[derive(Error, Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub enum MetaRaftError {
+    #[error(transparent)]
+    ForwardToLeader(#[from] ForwardToLeader),
+
+    #[error(transparent)]
+    ChangeMembershipError(#[from] ChangeMembershipError),
+
+    #[error("{0}")]
+    ConsistentReadError(String),
+
+    #[error("{0}")]
+    ClientWriteError(String),
```

Comment:
It seems there is only one error will be wrapped in `ClientWriteError`: `openraft::Fatal`.
What about naming it with `RaftFatal` or just `Fatal`?

BTW to wrap an external error, using `AnyError` with provide more info for debugging than using a `String`.